subscriptions/subscriptions-impl/src/main/java/com/duckduckgo/subscriptions/impl/ui/SubscriptionsWebViewActivity.kt

```diff
@@ -262,7 +267,10 @@ class SubscriptionsWebViewActivity : DuckDuckGoActivity(), DownloadConfirmationD
             renderPurchaseState(it.purchaseState)
         }.launchIn(lifecycleScope)
 
-        if (savedInstanceState == null && params.url == BUY_URL) {
+        val isPrivacyProUrl by lazy {
+            runCatching { subscriptions.isPrivacyProUrl(params.url.toUri()) }.getOrDefault(false)
+        }
+        if (savedInstanceState == null && isPrivacyProUrl) {
```

Comment:
Good catch! This WebView can be loaded for things other than the paywall, and the pixel was still being sent.